src/consensus.rs

```diff
@@ -302,7 +299,9 @@ impl <L, M> Consensus<L, M> where L: Log, M: StateMachine {
                     if latest_log_index < leader_prev_log_index {
                         scoped_debug!("AppendEntriesRequest: inconsistent previous log index: leader: {}, local: {}",
                                       leader_prev_log_index, latest_log_index);
-                        messages::append_entries_response_inconsistent_prev_entry(self.current_term())
+                        // Reply false if log doesn’t contain an entry at prevLogIndex whose term matches prevLogTerm
```

Comment:
it doesn't look like you're validating the term here, opposed to what the comment suggests. This is just the case in which the leader tries to append at a position which doesn't match up with the end of our log.